src/discounts/filters/provider.tsx

```diff
@@ -0,0 +1,29 @@
+import { ConditionalFilterContext } from "@dashboard/components/ConditionalFilter/context/context";
```

Comment:
The provider for product is within `ConditionalFilter/context/provider` so either move your provider to the same file, or move product one next to products